From 65d806e1ee3f288c3ce9eed367ae50c0dfcea0a1 Mon Sep 17 00:00:00 2001
From: Robert Marko <robert.marko@sartura.hr>
Date: Fri, 15 May 2026 11:26:25 +0200
Subject: [PATCH 2/2] net: sfp: add quirk for Dell M14MK

Like the BlueOptics BO27O856S1D-BO, Dell M14MK is a dual rate 25G/10G-SR
SFP28 module but it also does not advertise 10G-SR support in EEPROM.

So reuse the same "sfp_quirk_10g_25g_sr" to set the required bits for
10G-SR support.

Signed-off-by: Robert Marko <robert.marko@sartura.hr>
---
 drivers/net/phy/sfp.c | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

--- a/drivers/net/phy/sfp.c
+++ b/drivers/net/phy/sfp.c
@@ -537,9 +537,10 @@ static const struct sfp_quirk sfp_quirks
 
 	SFP_QUIRK_F("BIDB", "X-ONU-SFPP", sfp_fixup_potron),
 
-	// BlueOptics BO27O856S1D-BO is dual rate 25G/10G but the 10G rate is
-	// not advertised in EEPROM.
+	// BlueOptics BO27O856S1D-BO and DELL M14MK are dual rate 25G/10G but
+	// the 10G rate is not advertised in EEPROM.
 	SFP_QUIRK_S("BlueOptics", "BO27O856S1D-BO", sfp_quirk_10g_25g_sr),
+	SFP_QUIRK_S("DELL EMC", "M14MK", sfp_quirk_10g_25g_sr),
 
 	// FLYPRO SFP-10GT-CS-30M uses Rollball protocol to talk to the PHY.
 	SFP_QUIRK_F("FLYPRO", "SFP-10GT-CS-30M", sfp_fixup_rollball),
